app/questionnaire/router.py

```diff
@@ -116,9 +114,7 @@ def can_display_section_summary(
     ) -> bool:
         return bool(
             self._schema.get_summary_for_section(section_id)
-        ) and self._progress_store.is_section_or_repeating_blocks_progress_complete(
-            section_id, list_item_id
-        )
+        ) and self._progress_store.is_section_complete(section_id, list_item_id)
```

Comment:
How come we don't use the section key here? Looks like some more methods could be updated to use the new key? For example methods like `_get_first_incomplete_section_key` and `_get_last_complete_section_key`. Think there might be a few more too from my first check.

Reply:
I've added all I could find and it's a lot of extra modules refactored.